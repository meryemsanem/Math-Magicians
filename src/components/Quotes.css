.quotes-container {
  text-align: center;
}

.quote {
  font-size: 20px;
}

.loading {
  text-align: center;
  font-size: 30px;
}
