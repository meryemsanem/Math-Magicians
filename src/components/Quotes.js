import React, { useEffect, useState } from 'react';
import './Quotes.css';

function Quotes() {
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);
  const [quote, setQuote] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      try {
        setLoading(true);
        const res = await fetch(
          'https://api.api-ninjas.com/v1/quotes?category=learning',
          {
            headers: {
              'Content-Type': 'application/json',
              'X-API-key': 'y7y+TjrvdE97ECwfFUcgGQ==MXJnbsVOvq7sABp3',
            },
          },
        );
        const json = await res.json();
        setQuote(json);
      } catch (error) {
        setError(true);
      }
      setLoading(false);
    };
    fetchData();
  }, [setLoading, setError, setQuote]);

  if (loading) return <div className="loading">Quote is loading...</div>;
  if (error) return <div>Error!!!</div>;
  return (
    <div>
      {quote.map((item) => (
        <div key={item.id} className="quotes-container">
          <p className="quote">{item.quote}</p>
          <h1 className="author">{item.author}</h1>
        </div>
      ))}
    </div>
  );
}

export default Quotes;
