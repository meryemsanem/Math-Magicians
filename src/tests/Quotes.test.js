import React from 'react';
import {
  render, screen, waitFor, act,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Quotes from '../components/Quotes';

const waitForFetch = (data) => {
  global.fetch = jest.fn(() => Promise.resolve({
    json: () => Promise.resolve(data),
  }));
};

test('shows charging status while getting data', async () => {
  waitForFetch([]);

  render(<Quotes />);

  const loadingElement = await screen.findByText(/Quote is loading/i);
  expect(loadingElement).toBeInTheDocument();
});

test('show citations and authors after getting the data', async () => {
  waitForFetch([
    {
      id: 1,
      quote: 'Cita de ejemplo',
      author: 'Autor de ejemplo',
    },
  ]);

  await act(async () => {
    render(<Quotes />);
  });

  await waitFor(() => {
    const quoteElement = screen.getByText('Cita de ejemplo');
    const authorElement = screen.getByText('Autor de ejemplo');
    expect(quoteElement).toBeInTheDocument();
    expect(authorElement).toBeInTheDocument();
  });
});

test('matches the snapshot', () => {
  const componentTree = renderer.create(<Quotes />);
  expect(componentTree.toJSON()).toMatchSnapshot();
});
