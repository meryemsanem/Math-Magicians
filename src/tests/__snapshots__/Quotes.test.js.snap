// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Quotes coincide con el snapshot 1`] = `<div />`;

exports[`Quotes component matches snapshot 1`] = `<div />`;

exports[`coincide con el snapshot 1`] = `<div />`;

exports[`matches the snapshot 1`] = `<div />`;
